transactions/transactions_service.go

```diff
@@ -897,7 +897,13 @@ func (svc *transactionsService) markTransactionSettled(tx *gorm.DB, dbTransactio
 	}
 
 	if preimage == "" {
-		return nil, errors.New("no preimage in payment")
+		// outgoing payments MUST have a preimage
+		if dbTransaction.Type == constants.TRANSACTION_TYPE_OUTGOING {
+			return nil, errors.New("no preimage in payment")
+		}
+		// incoming payments SHOULD have a preimage
+		// (currently cashu backend does not)
```

Comment:
where does this apply? only cashu? why can transactions not have a preimage. 

is cashu worth this?

Reply:
I also wanted to add that not only are the transactions not showing up in history, I believe the following issue is due to the same cause (not having a preimage). 

When an invoice is created, in the Hub or Go, the GUI waits for payment indefinitely.

![image](https://github.com/user-attachments/assets/98e6a692-69c8-4297-9810-e2c6ce48d0d1)